gunicorn/arbiter.py

```diff
@@ -559,12 +557,8 @@ def kill_worker(self, pid, sig):
         try:
             os.kill(pid, sig)
         except OSError as e:
-            if e.errno == errno.ESRCH:
-                try:
-                    worker = self.WORKERS.pop(pid)
-                    worker.tmp.close()
-                    self.cfg.worker_exit(self, worker)
-                    return
-                except (KeyError, OSError):
-                    return
-            raise
+            if e.errno != errno.ESRCH:
+                raise
+
+        worker = self.WORKERS.pop(pid, None)
+        self.DYING_WORKERS.setdefault(pid, worker)
```

Comment:
Yeah, you're right. The error scenario there would be adding a useless key to `DYING_WORKERS` which would be a small memory leak. I'll fix that.